mongo/errors.go

```diff
@@ -340,7 +359,12 @@ func (we WriteError) Error() string {
 
 // HasErrorCode returns true if the error has the specified code.
 func (we WriteError) HasErrorCode(code int) bool {
-	return we.Code == code
+	return hasErrorCode(we, code)
```

Comment:
Re-implementing this in terms of `ErrorCodes` seems like an unnecessary change. The existing code is still valid and is more obvious.